server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -140,50 +138,131 @@ public CachingClusteredClient(
     );
   }
 
-  @Override
-  public Sequence<T> run(final QueryPlus<T> queryPlus, final Map<String, Object> responseContext)
+  private class SpecificQueryRunner
```

Comment:
This class contains the whole logic of CachingClusteredClient, so I don't want to move it out.

Added Javadoc and renamed to `SpecificQueryRunnable` to remove false association with QueryRunner interface.